ui/client/components/graph/node-modal/NodeDetailsContent.js

```diff
@@ -439,43 +444,50 @@ export class NodeDetailsContent extends React.Component {
     this.doCreateExpressionField(fieldName, fieldName, `${listFieldPath}.${expressionProperty}`, validators);
 
   doCreateExpressionField = (fieldName, fieldLabel, exprPath, validators) => {
-    const readOnly = !this.props.isEditMode;
-    const exprTextPath = `${exprPath}.expression`;
-    const expressionObj = _.get(this.state.editedNode, exprPath);
-    const isMarked = this.isMarked(exprTextPath);
-    const restriction = this.getRestriction(fieldName);
-    const showValidation = this.props.showValidation
+    const readOnly = !this.props.isEditMode
+    const exprTextPath = `${exprPath}.expression`
+    const expressionObj = _.get(this.state.editedNode, exprPath)
+    const isMarked = this.isMarked(exprTextPath)
+    const restriction = this.getRestriction(fieldName)
+    const {showValidation, showSwitch} = this.props
 
     if (restriction.hasFixedValues)
-      return <ExpressionWithFixedValues
-        fieldLabel={fieldLabel}
-        onChange={(newValue) => this.setNodeDataAt(exprTextPath, newValue)}
-        obj={expressionObj}
+      return (
+        <EditableExpression
+          fieldType={"expressionWithFixedValues"}
+          fieldLabel={fieldLabel}
+          expressionObj={expressionObj}
+          renderFieldLabel={this.renderFieldLabel}
+          values={restriction.values}
+          isMarked={isMarked}
+          showSwitch={showSwitch}
+          readOnly={readOnly}
+          onValueChange={(newValue) => this.setNodeDataAt(exprTextPath, newValue)}
+        />
+      )
+
+    const param = this.findParamByName(fieldLabel)
+    return TestRenderUtils.wrapWithTestResult(
```

Comment:
This should be a separeted highlevel component. Now it looks ugly :/

Reply:
Done, but not sure if we should refactor this in this PR